src/boot/efi/boot.c

```diff
@@ -2438,10 +2437,29 @@ static EFI_STATUS image_start(
         efivar_set_time_usec(LOADER_GUID, L"LoaderTimeExecUSec", 0);
         err = BS->StartImage(image, NULL, NULL);
         graphics_mode(FALSE);
```

Comment:
this loks misplaced. if we use the compat proto next, then we should only switch from graphics mode once that fails, too, no?

Reply:
This is deliberate to keep things simple. And it shouldn't really matter anyways.